components/prize_pool/commons/prize_pool.lua

```diff
@@ -578,4 +593,19 @@ function Placement:_parseOpponentArgs(input, date)
 	return Opponent.resolve(opponentData, date)
 end
 
+function Placement:_setUsdFromRewards(prizesToUse, prizeTypes)
+	for _, opponent in ipairs(self.opponents) do
+		if not opponent.prizeRewards[PRIZE_TYPE_USD .. 1] and not self.prizeRewards[PRIZE_TYPE_USD .. 1] then
+			local usdReward = 0
+			for _, prize in pairs(prizesToUse) do
+				local localMoney = opponent.prizeRewards[prize.id] or self.prizeRewards[prize.id]
+				if localMoney and localMoney > 0 then
+					usdReward = usdReward + prizeTypes[prize.type].convertToUsd(prize.data, localMoney, opponent.date)
+				end
+			end
+			opponent.prizeRewards[PRIZE_TYPE_USD .. 1] = Math.round{usdReward, 2}
+		end
```

Comment:
Make this a function and make the wrapper if an early return to avoid the heavy nesting

Reply:
Change to `Array.foreach` and `return` where otherwise continue would been used. Nesting a lot better now.